File: src/main/java/SpartJavaSample/AuthorizeController.java
package SpartJavaSample;

import spark.Request;
import spark.Response;

import java.util.Map;

public class AuthorizeController implements Controllable
{
    @Override
    public void process(Request request, Response response, final Map<String, Object> model)
    {
        model.put("Test", "test value");
    }
}

File: src/main/kotlin/SpartJavaSample/AuthorizeController.kt
package SpartJavaSample

import spark.Request
import spark.Response

public class AuthorizeController : Controllable() {
    public override fun before(request: Request, response: Response): Boolean
    {
        if (request.session(false) == null)
        {
            response.redirect("/login")

            return false
        }

        return true
    }

    public override fun get(request: Request, response: Response): ControllerResult = ControllerResult(model = mapOf(
            "user" to "jdoe",
            "data" to mapOf(
                    "e1" to "e1 value",
                    "e2" to "e2 value",
                    "e3" to "e3 value")))
}
